linode/manager
# refactor: M3-7034 - Event Messages Part 2

## Description 📝
Part 2 of the Event Messages refactor. This PR addresses primarily:
- Progress events
- Code cleanup
- UI updates

The aim of the PR is to deprecate `eventMessageGenerator_CMR`. The way it was always fetching Linode, Linode Types and Region data no matter what the event in order to hook some events and modify their message. With the new Event Messages factories we can now just build custom events as components and use them directly in factories actions so fetching is tightly to the action status.

## Changes  🔄
- Improve and optimize progress event logic
- Implement v2 versions of hooks and components (in order for the feature to remain behind a feature flag)
- Implement CDS guidelined (**UX approved** ) UI changes
  - notification pill color
  - if more than 9 unread notifications, show `9+`
  - new progress pill icon
  - unseen notification border
  - relative time, time remaining copy
- Close Notification Center when clicking on "View all events" since we land on the event page (UX approved as well)

💡 See the [self review](https://github.com/linode/manager/pull/10550/files/64550672f45505bc9e9b042bc2854ebfff685261) comments for more details

🗒️: the `time_remaining` value is inconsistently coming back from the API and being looked at. It should appear for a lot more progress events (ex, resize, migration) but is apparently currently broken. This is looked at in ARB-5127.

🧪: This PR's unit test coverage is pretty basic since it will be followed by a full e2e suite dedicated to events

## Preview 📷

**Events Page**
![Screenshot 2024-06-10 at 09 33 07](https://github.com/linode/manager/assets/130582365/de882b50-f661-4bcd-8e43-1dbe3e5c3b19)

**Notification Center**
![Screenshot 2024-06-10 at 17 52 18](https://github.com/linode/manager/assets/130582365/3b46779b-0099-4e72-96fc-51ae2995d6ee)

**Event Bell Notification**
![Screenshot 2024-06-10 at 17 09 16](https://github.com/linode/manager/assets/130582365/9172fdea-4f1a-448b-b024-7ab2fe3d223c)


## How to test 🧪

### Verification steps
(How to verify changes)
- Checkout PR and make sure the `eventMessagesV2` is turned on
- Check the of both the event page and the notification for a variety of progress events (boot, reboot, resize, migrate, clone etc)

## As an Author I have considered 🤔

*Check all that apply*
